plugins/hls-code-range-plugin/test/Ide/Plugin/CodeRangeTest.hs

```diff
@@ -50,5 +50,66 @@ testTree =
+            in [
+                -- General test
+                testCase "Test General Code Block" $ check
+                    (mkCodeRange (Position 1 1) (Position 5 10) [] CodeKindRegion)
+                    [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeRegion)],
+
+                -- Tests for code kind
+                testCase "Test Code Kind Region" $ check
+                    (mkCodeRange (Position 1 1) (Position 5 10) [] CodeKindRegion)
+                    [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeRegion)],
+                testCase "Test Code Kind Comment" $ check
+                    (mkCodeRange (Position 1 1) (Position 5 10) [] CodeKindComment)
+                    [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeComment)],
+                testCase "Test Code Kind Import" $ check
+                    (mkCodeRange (Position 1 1) (Position 5 10) [] CodeKindImports)
+                    [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeImports)],
+
+                -- Test for Code Portions with children
+                testCase "Test Children" $ check
+                    (mkCodeRange (Position 1 1) (Position 5 10) [
+                        mkCodeRange (Position 1 2) (Position 3 6) [] CodeKindRegion,
+                        mkCodeRange (Position 3 7) (Position 5 10) [] CodeKindRegion
+                    ] CodeKindRegion)
+                    [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeRegion),
+                        FoldingRange 1 (Just 2) 3 (Just 6) (Just FoldingRangeRegion),
+                        FoldingRange 3 (Just 7) 5 (Just 10) (Just FoldingRangeRegion)],
+
+                -- Single line
+                testCase "Test Single Line" $ check
+                    (mkCodeRange (Position 1 0) (Position 1 15) [] CodeKindRegion)
+                    [],
+
+                -- MultiLine imports
+                testCase "MultiLine Imports" $ check
+                    (mkCodeRange (Position 1 0) (Position 5 15) [] CodeKindImports)
+                    [FoldingRange 1 (Just 0) 5 (Just 15) (Just FoldingRangeImports)]
+            ],
+
+        testGroup "createFoldingRange" $
+        let check :: CodeRange -> Maybe FoldingRange -> Assertion
+            check codeRange = (createFoldingRange codeRange @?=)
+
+            mkCodeRange :: Position -> Position -> V.Vector CodeRange -> CodeRangeKind -> CodeRange
+            mkCodeRange start end children crk = CodeRange (Range start end) children crk
+        in [
+            -- General test
+            testCase "Test General Code Block" $ check
+                (mkCodeRange (Position 1 1) (Position 5 10) [] CodeKindRegion)
+                (Just (FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeRegion))),
+            -- General test
+            testCase "Test Same Start Line" $ check
+                (mkCodeRange (Position 1 1) (Position 1 10) [] CodeKindRegion)
+                Nothing
```

Comment:
Why is this Nothing? I think this would be answered by more explanation about what you're doing with code ranges on the same line

Reply:
So, Something in one line does not need folding like what will it exactly fold. So I am removing all the folding ranges that start and end on the same line. (This logic is also handled by vscode itself, double proofing it by removing them).

I'll add an explanation about this in the comments.